File: read.m
%Example to read the data
%Created on Thu Mar 12 10:26:25 2026
%
%@author: max scharf

path = "data.json";
txt  = fileread(path);
data = jsondecode(txt);


% --- convert srt -> 45x11 numeric matrix X (with NaN for missing) ---
Xcell = cellfun(@toDoubleRowMissing, data.data, 'UniformOutput', false); % each -> 1x11 double
X     = vertcat(Xcell{:});                                         % 45x11 double

% --- build table with named columns and rows ---
T = array2table(X, 'VariableNames', data.column_names(:).', 'RowNames', data.row_names);

% optional: ensure valid MATLAB variable names
T.Properties.VariableNames = matlab.lang.makeValidName(T.Properties.VariableNames);

disp(T)

% ===== local functions =====
function r = toDoubleRowMissing(v)
    if iscell(v)
        r = cellfun(@(x) oneVal(x), v(:)).';
    else
        r = arrayfun(@(x) double(x), v(:)).';
    end
end

function y = oneVal(x)
    % Map "missing value" -> NaN; otherwise numeric conversion
    if isstring(x) || ischar(x)
        sx = string(x);
        if sx == "missing value"
            y = NaN;
        else
            y = str2double(sx);
        end
    else
        y = double(x);
    end
end

File: read.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Example to read the data
Created on Thu Mar 12 10:26:25 2026

@author: max scharf
"""

import json
import numpy as np

path = "data.json"

with open(path, "r", encoding="utf-8") as f:
    data = json.load(f)   # parses JSON into Python dictionary
srt = data["data"]
row = data["row_names"]
col = data["column_names"]

# %%
# prepare data format
srt_markedMissing = [
    [np.nan if x == "missing value" else np.double(x) for x in row] for row in srt]

# %%
# print in a neat matrix
print(("{:>10}"*(len(col)+1)).format("ID", *col))
for i, r in enumerate(srt_markedMissing):
    print(("{:>10}"+"{:10.2f}"*(len(r))).format(row[i], *r))
